src/Microsoft.ML.Transforms/Text/TextStaticExtensions.cs

```diff
@@ -113,4 +115,466 @@ public bool Equals(Reconciler other)
+            int skipLength = 0,
+            bool allLengths = true,
+            uint seed = 314489979,
+            bool ordered = true,
+            int invertHash = 0) => new OutPipelineColumn(input, hashBits, ngramLength, skipLength, allLengths, seed, ordered, invertHash);
+    }
+
+    /// <summary>
+    /// Extensions for statically typed ngram estimator.
+    /// </summary>
+    public static class NgramEstimatorExtensions
+    {
+        private sealed class OutPipelineColumn : Vector<float>
+        {
+            public readonly VarVector<Key<uint, string>> Input;
+
+            public OutPipelineColumn(VarVector<Key<uint, string>> input, int ngramLength, int skipLength, bool allLengths, int maxNumTerms, NgramTransform.WeightingCriteria weighting)
+                : base(new Reconciler(ngramLength, skipLength, allLengths, maxNumTerms, weighting), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler, IEquatable<Reconciler>
+        {
+            private readonly int _ngramLength;
+            private readonly int _skipLength;
+            private readonly bool _allLengths;
+            private readonly int _maxNumTerms;
+            private readonly NgramTransform.WeightingCriteria _weighting;
+
+            public Reconciler(int ngramLength, int skipLength, bool allLengths, int maxNumTerms, NgramTransform.WeightingCriteria weighting)
+            {
+                _ngramLength = ngramLength;
+                _skipLength = skipLength;
+                _allLengths = allLengths;
+                _maxNumTerms = maxNumTerms;
+                _weighting = weighting;
+
+            }
+
+            public bool Equals(Reconciler other)
+            {
+                return _ngramLength == other._ngramLength &&
+                _skipLength == other._skipLength &&
+                _allLengths == other._allLengths &&
+                _maxNumTerms == other._maxNumTerms &&
+                _weighting == other._weighting;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                Contracts.Assert(toOutput.Length == 1);
+
+                var pairs = new List<(string inputs, string output)>();
+                foreach (var outCol in toOutput)
+                    pairs.Add((inputNames[((OutPipelineColumn)outCol).Input], outputNames[outCol]));
+
+                return new NgramEstimator(env, pairs.ToArray(), _ngramLength, _skipLength, _allLengths, _maxNumTerms, _weighting);
+            }
+        }
+
+        /// <summary>
+        /// Produces a bag of counts of ngrams (sequences of consecutive words ) in a given text.
```

Comment:
>text [](start = 92, length = 4)

it's not a text. Mention that it typically is applied to the output of tokenizers